dataflow/extensible-templates/dataflow_udf_transform.js

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START dataflow_extensible_template_udf]
+/**
+ * User-defined function (UDF) to transform events
+ * as part of a Dataflow template job.
+ *
+ * @param {string} inJson input Pub/Sub JSON message (stringified)
+ */
+ function process(inJson) {
+    var obj = JSON.parse(inJson),
+        includePubsubMessage = obj.data && obj.attributes,
```

Comment:
Nit: I found the use of commas a little confusing to define multiple variables in the same statement since I didn't know that feature. It might be easier to read if we explicitly define each variable in their own statement. I'll leave that to you as personal style, I don't have a strong opinion.